packages/engine/errors/2327.md

```diff
@@ -0,0 +1,19 @@
+---
+original: "Property '{0}' is optional in type '{1}' but required in type '{2}'."
+excerpt: "Property '{0}' in type '{2}' cannot be 'undefined'."
+---
+
+You're assigning to a variable whose type `A` matches `B` but `B` has a property which can be `undefined` whereas `A` disallows it. For example:
```

Comment:
I'd like somewhere in these explanations to have 1 and 2 to be displayed in full (this is where prettifying could come in handy)